build/Makefile

```diff
@@ -59,7 +59,7 @@ KIND_CONTAINER_NAME=$(KIND_PROFILE)-control-plane
 # Game Server image to use while doing end-to-end tests
 GS_TEST_IMAGE ?= gcr.io/agones-images/udp-server:0.21
 
-ALPHA_FEATURE_GATES ?= "PlayerTracking=true"
+ALPHA_FEATURE_GATES ?= "PlayerTracking=true&GameServerCaching=true"
```

Comment:
Not a huge fan of "GameServerCaching" as a name Feels a bit ambiguous as to it's meaning. Maybe something that has "SDK" in it?

Some ideas:
- SDKWatchSendImmeadiate
- SDKWatchSendOnExecute
- WatchGameServerImmeadiate

Something like that? Not sure I like any of them 100% more, but something in that area would probably be good. WDYT?

As I discovered the other day, this will also need to be added to the cloudbuild.yaml
https://github.com/googleforgames/agones/blob/master/cloudbuild.yaml#L234

Reply:
Agree. It feels like game server caching is being used to achieve the feature of sending on watch. Could actually implement that without any caching and might want to consider as separate features.
Would throw in feature suggestions of `SendGameServerOnWatch` or `SendOnWatch` as the context of what is being watched can be gleaned from the SDK function WatchGameServer, and this could be extended to also send immediately if we add any future watchers.